server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeValueHealth.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.awarenesshealth;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.WeightedRouting;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Cluster Awareness AttributeValue Health information
+ */
+public class ClusterAwarenessAttributeValueHealth implements Writeable, ToXContentFragment {
+
+    private static final String ACTIVE_SHARDS = "active_shards";
+    private static final String INITIALIZING_SHARDS = "initializing_shards";
+    private static final String RELOCATING_SHARDS = "relocating_shards";
+    private static final String UNASSIGNED_SHARDS = "unassigned_shards";
+    private static final String NODES = "data_nodes";
+    private static final String WEIGHTS = "weight";
+    private final String name;
+    private int activeShards;
+    private int unassignedShards;
+    private int initializingShards;
+    private int relocatingShards;
+    private int nodes;
+    private double weight;
+    private List<String> nodeList;
+
+    /**
+     * Creates Awareness AttributeValue Health information
+     *
+     * @param name name of awareness attribute
+     */
+    public ClusterAwarenessAttributeValueHealth(String name) {
+        this.name = name;
+        this.nodeList = new ArrayList<>();
+    }
+
+    // Constructor use by Unit test case.
+    public ClusterAwarenessAttributeValueHealth(
+        String name,
+        int activeShards,
+        int initializingShards,
+        int relocatingShards,
+        int unassignedShards,
+        int nodes,
+        double weights
+    ) {
+        this.name = name;
+        this.activeShards = activeShards;
+        this.initializingShards = initializingShards;
+        this.relocatingShards = relocatingShards;
+        this.unassignedShards = unassignedShards;
+        this.nodes = nodes;
+        this.weight = weights;
+    }
```

Comment:
Better avoid ctors just for tests, make these package private in any case

Reply:
This constructor is used by unit test cases for some serialization tests as well as to test few functionalities.